crates/trainer/src/main.rs

```diff
@@ -15,25 +18,62 @@ fn main() {
     // 4. Mutate children
     // 5. Repeat
 
+    let mut last_saved = None;
+    let mut model = ThinkingLayer::new(
+        PongGame::input_nodes(),
+        PongGame::input_nodes() + 5 + PongGame::output_nodes(),
+        PongGame::output_nodes(),
+        Tanh,
+    )
+    .unwrap();
+    if let Ok(bytes) = fs::read("model.json") {
+        println!("Loading model...");
+        let data: PersistedGenome = serde_json::from_slice(&bytes).unwrap();
+        model.load_genome(data.genome);
+        last_saved = Some(data.score);
+    }
+
     let mut trainer = ModelTrainer::new(
-        ThinkingLayer::new(
-            PongGame::input_nodes(),
-            6 + 5,
-            PongGame::output_nodes(),
-            Tanh,
-        )
-        .unwrap(),
+        model,
         TrainConfig {
             epoch_size: 500,
             sample_size: 10,
             survival_rate: 0.1,
-            mutation_probability: 0.02,
-            mutation_strength: 0.05,
+            mutation_probability: 0.05,
+            mutation_strength: 0.2,
         },
     );
 
-    for generation_index in 0..10000 {
+    for generation_index in 0..100 {
         trainer.train_next_gen::<PongGame>();
+        let all_time_best = trainer.overall_best().as_ref().unwrap().score;
+        let genome = trainer.overall_best().as_ref().unwrap().model.genome();
+
+        match &last_saved {
+            None => {
+                println!("Saving new best model...");
+                last_saved = Some(all_time_best);
+                let json = serde_json::to_string(&PersistedGenome {
+                    score: all_time_best,
+                    genome: genome.to_vec(),
+                })
+                .unwrap();
+                fs::write("model.json", json).unwrap();
+            }
+            Some(last_value) => {
+                if *last_value < all_time_best {
+                    println!("Saving new best model...");
+                    last_saved = Some(all_time_best);
+                    let json = serde_json::to_string(&PersistedGenome {
+                        score: all_time_best,
+                        genome: genome.to_vec(),
+                    })
+                    .unwrap();
+                    fs::write("model.json", json).unwrap();
+                }
+            }
+        }
```

Comment:
Could be done via last_saved.unwrap_or_default(0) or whatever the function is called so you don't have to duplicate the code as much.